validation/genesis/genesis-allocs_test.go

```diff
@@ -0,0 +1,171 @@
+package genesis
+
+import (
+	"encoding/json"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	. "github.com/ethereum-optimism/superchain-registry/validation/common"
+
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/stretchr/testify/require"
+)
+
+var temporaryOptimismDir string
+
+// TestMain is the entry point for testing in this package.
+func TestMain(m *testing.M) {
+	// Clone optimism into gitignored temporary directory (if that directory does not yet exist)
+	// We avoid cloning under the superchain-registry tree, since this causes dependency resolution problems
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		panic("No caller information")
+	}
+	thisDir := filepath.Dir(filename)
+	temporaryOptimismDir = path.Join(thisDir, "../../../optimism-temporary")
+
+	// Clone the repo if it the folder doesn't exist
+	_, err := os.Stat(temporaryOptimismDir)
+	needToClone := os.IsNotExist(err)
+	if needToClone {
+		mustExecuteCommandInDir(thisDir,
+			exec.Command("git", "clone", "https://github.com/ethereum-optimism/optimism.git", temporaryOptimismDir))
+	}
+
+	// Run tests
+	exitVal := m.Run()
+
+	// Teardown code:
+	// Only if we cloned the directory, now delete it
+	// This means during local development, one can clone the
+	// repo manually before running the test to speed up runs.
+	if needToClone {
+		if err := os.RemoveAll(temporaryOptimismDir); err != nil {
+			panic("Failed to remove temp directory: " + err.Error())
+		}
+	}
+
+	// Exit with the result of the tests
+	os.Exit(exitVal)
+}
+
+func TestGenesisAllocs(t *testing.T) {
+	for _, chain := range OPChains {
+		if chain.SuperchainLevel == Standard || chain.StandardChainCandidate {
+			t.Run(PerChainTestName(chain), func(t *testing.T) {
+				// Do not run in parallel
```

Comment:
Might be good to explain why we don't want to run in parallel. Assuming its because we don't want to checkout a different version of the monorepo while another test is running